src/FundraiserMailingGenerator/components/ButtonCreator.tsx

```diff
@@ -62,8 +64,10 @@ export default class ButtonCreator extends Component<Props, State> {
   // TODO: Refactor this. At the moment it's being duplicated in most
   // components but we should extract it.
   build = (): string => {
-    const { url, rates, lang, correctLowAsks } = this.props;
+    const { url, rates, lang, correctLowAsks, recurringDefault } = this.props;
     const template = this.state.template[this.props.lang];
+    const recurringDef =  (recurringDefault && recurringDefault.length < 2 || !recurringDefault) ? undefined : recurringDefault;
```

Comment:
@Tuuleh 
If params comes as `recurring_default=' with  ( empty value),  I reject that

Reply:
Why don't you just check that it's one of the values that are valid? `'recurring'`, `'only_recurring'`, `'one_off'` or `'only_one_off'`

Reply:
@Tuuleh 
I am passing empty value as well in the "recurring_default"  select. 

 ```
<select name="recurringDefault" className='select' onChange={this.handleChange }>
            <option value="">Default</option>
            <option value="only_one_off">Only One-off Donation </option>
            <option value="only_recurring">Only Recurring Donation</option>
          </select>
```

Reply:
Okay, then I'm not sure your check is necessary, because the generator does pas an empty value for `recurring_default` if the page default is selected. Example: 
https://feature.sumofus.org/a/donate?ak_proof=1&akid=.15735918.-30FP6&amount=1&currency=EUR&rd=1&recurring_default=&source=fwd&t=12
It doesn't break the functionality, so maybe this check is overall redundant?